viewing_party/party.py

```diff
@@ -1,23 +1,151 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if title == None or genre == None or rating == None:
+        movie_data = None
+
+    else:
+        movie_data = {
+            'title': title,
+            'genre': genre,
+            'rating': rating
+            }
+
+    return movie_data
+
+
+def add_to_watched(user_data, movie):
+    user_data['watched'].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data['watchlist'].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, MOVIE_TITLE_1):
+    for movies in user_data.values():
+        for movie in movies:
+            if movie["title"] == MOVIE_TITLE_1:
+                add_movie = movies.pop(movies.index(movie))

```

Comment:
We are continuing to iterate after removing a value from a mutable list. This puts us in _the danger zone_ we discussed in the "Iterating over data" roundtable. After an item is removed from a list that is being looped over, the iterator that gives us the next element for our loop will likely not be pointing where we expect! Some approaches to handle this could be creating a deep copy of the data before we loop and editing the copy instead of the original we are looping over, or by breaking out of the loop after we remove an item (so we do not continue iterating over the altered data).